### PR TITLE
QOL Flakes Improvements

### PR DESCRIPTION
**Pull request by JonathanLorimer**
This PR adds:
- `flake-utils` for cross platform compatibility
- `pre-commit-hooks.nix` for formatting and linting
- `github actions ci` - A minimal CI config to make things easier when we want to build this out
- slight updates to the `README.md`

### CODE REVIEW

**Comment by JonathanLorimer** [comment]
I couldn't test the Github Actions, I think an admin needs to enable them. I _think_ they should just work.

File: .github/workflows/ci.yml
Comment (soareschen) [inline .github/workflows/ci.yml:17]: I think we don't need this as we don't use any private repository.

File: .github/workflows/ci.yml
Comment (soareschen) [inline .github/workflows/ci.yml:19]: We can leave this out of the commit for now.

File: .gitignore
Comment (soareschen) [inline .gitignore:2]: Is this file being added in this PR?

**Review by soareschen — APPROVED** [review]
Thanks! Looks good other than some nit. Feel free to merge after the update. We can test on the CI after merging.

Btw FYI I will be having vacation tomorrow onward until Monday, so I might not respond to new PR for a while.

File: .gitignore
Comment (JonathanLorimer) [reply .gitignore:2]: This file gets generated once you enter the nix-shell, and it links to a file in the `nix-store`, so its best to git ignore it!

Comment (soareschen) [reply .gitignore:2]: Got it. Alright then!

**Review by soareschen — APPROVED** [review]

**Comment by JonathanLorimer** [comment]
@soareschen I believe you have to merge the PR, or otherwise change branch protection rules so that non-admins can merge with N approvals.

**Comment by soareschen** [comment]
Didn't know that. I have added you as a maintainer of the repository now so you should be able to push to the repository directly.

**Pull request merged**
